wp-includes/js/service-worker.js

```diff
@@ -130,6 +130,63 @@ wp.serviceWorker = workbox;
 			super.registerRoute( route );
 			return route;
 		}
+
+		/**
+		 * Register a route that will return a precached file for a navigation
+		 * request. This is useful for the
+		 * [application shell pattern]{@link https://developers.google.com/web/fundamentals/architecture/app-shell}.
+		 *
+		 * This method will generate a
+		 * [NavigationRoute]{@link workbox.routing.NavigationRoute}
+		 * and call
+		 * [Router.registerRoute()]{@link workbox.routing.Router#registerRoute}
+		 * .
+		 *
+		 * @param {string} cachedAssetUrl
+		 * @param {Object} [options]
+		 * @param {string} [options.cacheName] Cache name to store and retrieve
+		 * requests. Defaults to precache cache name provided by
+		 * [workbox-core.cacheNames]{@link workbox.core.cacheNames}.
+		 * @param {Array<RegExp>} [options.blacklist=[]] If any of these patterns
+		 * match, the route will not handle the request (even if a whitelist entry
+		 * matches).
+		 * @param {Array<RegExp>} [options.whitelist=[/./]] If any of these patterns
+		 * match the URL's pathname and search parameter, the route will handle the
+		 * request (assuming the blacklist doesn't match).
+		 * @return {workbox.routing.NavigationRoute} Returns the generated
+		 * Route.
+		 *
+		 * @alias workbox.routing.registerNavigationRoute
+		 */
+		registerNavigationRoute(cachedAssetUrl, options = {}) {
```

Comment:
This is copied straight from Workbox: https://github.com/GoogleChrome/workbox/blob/912080a1bf3255c61151ca3d0ebd0895aaf377e2/packages/workbox-routing/_default.mjs#L126-L194

Copying is needed unfortunately because we fork the Routing logic in order to customize how the route matching is done.